Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -593,14 +593,17 @@ bool RenderLayerCompositor::updateCompositingPolicy()
         m_compositingPolicy = page().compositingPolicyOverride().value();
         return m_compositingPolicy != currentPolicy;
     }
-    
+
+    const auto isCurrentlyUnderMemoryPressureOrWarning = [] {
+        return MemoryPressureHandler::singleton().isUnderMemoryPressure() || MemoryPressureHandler::singleton().isUnderMemoryWarning();
+    };
+
     static auto cachedMemoryPolicy = WTF::MemoryUsagePolicy::Unrestricted;
-    static MonotonicTime cachedMemoryPolicyTime;
-    static constexpr auto memoryPolicyCachingDuration = 2_s;
-    auto now = MonotonicTime::now();
-    if (now - cachedMemoryPolicyTime > memoryPolicyCachingDuration) {
+    static auto cachedIsUnderMemoryPressureOrWarning = isCurrentlyUnderMemoryPressureOrWarning();
+
+    if (cachedIsUnderMemoryPressureOrWarning != isCurrentlyUnderMemoryPressureOrWarning()) {
         cachedMemoryPolicy = MemoryPressureHandler::singleton().currentMemoryUsagePolicy();
-        cachedMemoryPolicyTime = now;
+        cachedIsUnderMemoryPressureOrWarning ^= true;
```

Comment:
I think you could avoid this confusing `^=` by writing this as:
```
bool nowUnderMemoryPressure = isCurrentlyUnderMemoryPressureOrWarning();
static bool isUnderMemoryPressureOrWarning = nowUnderMemoryPressure;

if (isUnderMemoryPressureOrWarning != nowUnderMemoryPressure) {
  cachedMemoryPolicy = MemoryPressureHandler::singleton().currentMemoryUsagePolicy();
  isUnderMemoryPressureOrWarning = nowUnderMemoryPressure;
}
```

Reply:
Cool, I like it. Though I don't think bit flips are all that confusing yours is definitely more readable.